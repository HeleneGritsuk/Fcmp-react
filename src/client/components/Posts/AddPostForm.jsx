import React from 'react';
import styles from '../../styles/addPost.css.js';

class AddPostForm extends React.Component {

  render() {
    let author;
    let textInput;
    let postTitle;

    return (
      <div style={styles.formWrapper}>
        <div style={styles.formHeader}>
          <div style={styles.inputWrapper}>
            <div>
                            Author:</div>
            <input ref={node => {
              author = node;
            }} style={styles.input}
            />
          </div>
          <div style={styles.inputWrapper}>
            <div>
                            Title:</div>
            <input ref={node => {
              postTitle = node;
            }} style={styles.input}
            />
          </div>
        </div>
        <div style={styles.textAreaWrapper}>
          <div>
                        Text:</div>
          <textarea ref={node => {
            textInput = node;
          }} style={styles.textArea}
          />
        </div>
        <div>
          <button onClick={() => {
            this.props.onClick(author.value, textInput.value, postTitle.value);
            author.value = '';
            textInput.value = '';
            postTitle.value = '';
          }}
          >

            Add Post
          </button>
        </div>
      </div>
    );
  }

}

export default AddPostForm;
