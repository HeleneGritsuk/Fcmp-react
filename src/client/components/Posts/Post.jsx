import React from 'react';
import styles from '../../styles/postsList.css.js';

class Post extends React.Component {

  render() {
    const {  author,  text,   title } = this.props;

    return (
      <div style = {styles.postContainer}>
        <div style = {styles.titleStyle}>{title}</div>
        <div style = {styles.authorWrapper}>by {author}</div>
        <div>Text: {text}</div>
      </div>
    );
  }
}
export default Post;
