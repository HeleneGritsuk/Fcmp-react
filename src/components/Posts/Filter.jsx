import React from 'react';
import styles from '../../styles/FilterBlock.css.js';

class Filter extends React.Component {

  render() {
    let searchInput;

    return (
      <div style = {styles.searchWrapper}>
        Search:
        {' '}
        <input style = {styles.searchInput}
          ref={node => {
            searchInput = node;
          }}
          onChange={
            () => this.props.onInputChange(searchInput.value)
          }
        />
      </div>
    );
  }

}

export default Filter;
