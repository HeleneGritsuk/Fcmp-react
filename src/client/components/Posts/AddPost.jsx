import React from 'react';
import { connect } from 'react-redux';

import { manualBlogSend } from '../../redux/actions';

import AddPostForm from './AddPostForm.jsx';

class AddPost extends React.Component {
  render() {
    const { onClick } = this.props;

    return (
      <AddPostForm onClick = {onClick} />
    );
  }
}

const mapDispatchProps = (dispatch) => {
  return {
    onClick: (author, text, title) => {
      dispatch(manualBlogSend({ author, text, title }));
    }
  };
};

export default connect(null, mapDispatchProps)(AddPost);
