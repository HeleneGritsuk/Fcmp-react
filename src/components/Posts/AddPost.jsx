import React from 'react';
import { connect } from 'react-redux';

import { addPost } from '../../redux/actions';

import AddPostForm from './AddPostForm.jsx';

class AddPost extends React.Component {
  render() {
    const { onClick } = this.props;

    return (
      <AddPostForm onClick = {onClick} />
    );
  }
}

const mapDispatchProps = (dispatch) => {
  return {
    onClick: (author, textInput, postTitle) => {
      dispatch(addPost(author, textInput, postTitle));
    }
  };
};

export default connect(null, mapDispatchProps)(AddPost);
