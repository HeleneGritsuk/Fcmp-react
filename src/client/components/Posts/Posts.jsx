import React from 'react';
import styles from '../../styles/postsList.css.js';
import Post from './Post.jsx';

class Posts extends React.Component {

  render() {
    const { posts } = this.props;

    return (
      <ul style = {styles.postsWrapper}>
        {
          posts.map(post =>
            <Post
              key={post.id}
              {...post}
            />
        )}
      </ul>
    );
  }

}

export default Posts;
