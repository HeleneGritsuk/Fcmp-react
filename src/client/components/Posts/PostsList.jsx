import React from 'react';
import { connect } from 'react-redux';


import Posts from './Posts.jsx';
import { getAllPosts } from '../../redux/actions';


class PostList extends React.Component {
  componentDidMount() {

    this.props.getAllPosts();
  }

  render() {

    const { posts } = this.props;

    return (
      <Posts posts = {posts}/>
    );
  }
}


const getVisiblePosts = (
  posts,
  filter
) => {
  switch (filter) {
    case '':
      return posts;
    default:
    debugger;
      return posts.filter((item) => {
        return item.author.toLowerCase().search(filter) !== -1;
      });
  }
};

const mapStateToProps = (
  state
) => {
  return {
    posts: getVisiblePosts(
      state.posts,
      state.visibilityFilter
    ),
    filter: state.visibilityFilter
  };
};

const mapDispatchProps = (dispatch) => {
  return {
    getAllPosts: () => {
      dispatch(getAllPosts());
    }
  };
};


export default connect(
  mapStateToProps,
  mapDispatchProps
)(PostList);
