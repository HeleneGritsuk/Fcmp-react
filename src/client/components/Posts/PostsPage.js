import React from 'react';
import PostsList from './PostsList.jsx';
import FilterBlock from './FilterBlock.jsx';
import AddPost from './AddPost.jsx';

class PostsPage extends React.Component {
  render() {
    return (
      <div>
        <FilterBlock />
        <AddPost />
        <PostsList />
      </div>
    );
  }
}

export default PostsPage;
