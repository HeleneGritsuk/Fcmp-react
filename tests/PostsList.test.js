import React from 'react';
import Posts from '../src/client/components/Posts/Posts.jsx';
import renderer from 'react-test-renderer';

const mockPostsList = [
  {
    author: "THE ASSOCIATED PRESS",
    id: "5ac1134b9bc42c482c71e594",
    text: "China's defunct Tiangong 1 space station is hurtling toward Earth and expected to re-enter the atmosphere within the next day.\nMost of it should burn up on re-entry, so scientists say it poses only a â€¦",
    title: "Defunct Chinese space lab set to re-enter Earth's atmosphere1"
  },
  {
    author: "ABC News",
    id: "5ac113689bc42c482c71e595",
    text: "The computer expert who alleges a trove of Facebook data was improperly used to help Donald Trump's White House bid says he strongly believes the",
    title: "Defunct Chinese space lab set to re-enter Earth's atmosphere1"
  }
]
describe('PostsList', () => {
  it('should render list of Posts correctly', () => {
    const tree = renderer.create(<Posts posts = {mockPostsList} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
