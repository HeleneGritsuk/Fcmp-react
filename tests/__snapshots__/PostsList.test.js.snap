// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsList should render list of Posts correctly 1`] = `
<ul
  style={
    Object {
      "borderTop": "1px solid gray",
      "padding": "0",
      "paddingTop": "40px",
    }
  }
>
  <div
    style={
      Object {
        "border": "1px solid #dadada",
        "borderRadius": "5px",
        "marginBottom": "15px",
        "padding": "15px",
        "width": "300px",
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
          "textTransform": "uppercase",
        }
      }
    >
      Defunct Chinese space lab set to re-enter Earth's atmosphere1
    </div>
    <div
      style={
        Object {
          "color": "#8d8d8d",
          "fontSize": "12px",
          "fontStyle": "italic",
          "textAlign": "right",
        }
      }
    >
      by 
      THE ASSOCIATED PRESS
    </div>
    <div>
      Text: 
      China's defunct Tiangong 1 space station is hurtling toward Earth and expected to re-enter the atmosphere within the next day.
Most of it should burn up on re-entry, so scientists say it poses only a â€¦
    </div>
  </div>
  <div
    style={
      Object {
        "border": "1px solid #dadada",
        "borderRadius": "5px",
        "marginBottom": "15px",
        "padding": "15px",
        "width": "300px",
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
          "textTransform": "uppercase",
        }
      }
    >
      Defunct Chinese space lab set to re-enter Earth's atmosphere1
    </div>
    <div
      style={
        Object {
          "color": "#8d8d8d",
          "fontSize": "12px",
          "fontStyle": "italic",
          "textAlign": "right",
        }
      }
    >
      by 
      ABC News
    </div>
    <div>
      Text: 
      The computer expert who alleges a trove of Facebook data was improperly used to help Donald Trump's White House bid says he strongly believes the
    </div>
  </div>
</ul>
`;
